PyAssembler/Assembler.py

```diff
@@ -0,0 +1,326 @@
+            for label_tuple in self.sym_tbl:
+                damaged_line = damaged_line.replace(label_tuple, str(self.sym_tbl[label_tuple]))
+            
+            damaged_line = eval(damaged_line)
+            return format(int(damaged_line) & 0xffff, '016b')
+
+
+    def section_handler(self, words, translated_words):
+        
+        if words[0] == Sections.STRING:
+            translated_words[0] = self.convert_str_to_binary(words[1].strip('"'))       # remove " "
+        
+        elif words[0] == Sections.DATA:
+            self.current_address += 2
+            translated_words[0] = format(int(words[1]), '016b')
+        
+        else:
+            return True
+
+
+    def clean_comments(self, asm_lines):
+
+        for index, line in enumerate(asm_lines):
+            line = line.strip()
+            
+            # Blank lines or Comment lines
+            if line == '' or line.startswith(';'):
+                asm_lines[index] = None
+            
+            else:
+                # Inline comments
+                code_then_comment = line.split(';')
+                asm_lines[index] = code_then_comment[0].strip()
+        
+        return [line for line in asm_lines if line is not None]
+
+
+    def first_step(self, assembly_file):
+        
+        with open(assembly_file,'r') as origin_file:
+            all_lines = origin_file.readlines()
+        
+        cleaned_lines = self.clean_comments(all_lines)
+        
+        # Translates the assembly lines with place-holders for the yet unconvertable expressions
+        semi_trans_lines = [self.translate_line(line) for line in cleaned_lines]
+        
+        # Separate adjacent lines
+        semi_trans_lines = [line.split('\n') for line in semi_trans_lines]
+        semi_trans_lines = [element for sublist in semi_trans_lines for element in sublist]
+        
+        return semi_trans_lines
+
+
+    def second_step(self, outfile_name, semi_trans_lines):
+
+        # Fix lines using symbol table & place-holders
+        all_lines = [self.fix_line(line.strip()) for line in semi_trans_lines]
+        
+        # Prepare binary writing
+        output_binary_string = ''.join(all_lines)
+        bytes_string = [(output_binary_string[i:i+8]) for i in range(0, len(output_binary_string), 8)]
+        
+        with open(outfile_name, 'wb') as bin_out:
+            for byte_str in bytes_string:
+                bin_out.write(struct.pack('>B', int(byte_str, 2)))
+
+
+    def assemble_code(self):
+
+        half_translated_lines = self.first_step(self.input_file)
+        
+        self.second_step(self.output_file, half_translated_lines)
+        
+        print("Assembling Done.")
+
+
+
+def setup_argparse():
+    parser = argparse.ArgumentParser()
+    
+    parser.add_argument("-i", "--input", help="Directs the input to a name of your choice")
+    parser.add_argument("-o", "--output", help="Directs the output to a name of your choice")

```

Comment:
Flags are optional by default. These should either be positional (no `-`) or specify `required=True`.